extensions/amp-subscriptions/amp-subscriptions.md

```diff
@@ -47,160 +74,366 @@ Because of standardization of markup, support for multiple providers, and improv
+    <td>The Source URL of this AMP document. If the document is served from a CDN, the AMPDOC_URL will be a CDN URL, while SOURCE_URL will be the original source URL.</td>
+  </tr>
+  <tr>
+    <td class="col-thirty"><code>AMPDOC_URL</code></td>
+    <td>The URL of this AMP document.</td>
+  </tr>
+  <tr>
+    <td class="col-thirty"><code>CANONICAL_URL</code></td>
+    <td>The canonical URL of this AMP document.</td>
+  </tr>
+  <tr>
+    <td class="col-thirty"><code>DOCUMENT_REFERRER</code></td>
+    <td>The Referrer URL.</td>
+  </tr>
+  <tr>
+    <td class="col-thirty"><code>VIEWER</code></td>
+    <td>The URL of the AMP Viewer.</td>
+  </tr>
+  <tr>
+    <td class="col-thirty"><code>RANDOM</code></td>
+    <td>A random number. Helpful to avoid browser caching.</td>
+  </tr>
+</table>
+
+Here’s an example of the URL extended with Reader ID, Canonical URL, Referrer information and random cachebuster:
+
+```http
+https://pub.com/amp-authorization?
+   rid=READER_ID
+  &url=CANONICAL_URL
+  &ref=DOCUMENT_REFERRER
+  &_=RANDOM
+```
 
-Using Microdata, the markup could look like this:
+`AUTHDATA` variable is available to Pingback and Login URLs. It allows passing any field in the authorization
+response as an URL parameter. E.g. `AUTHDATA(data.isLoggedIn)`.
 
-```html
-<div itemscope itemtype="http://schema.org/NewsArticle">
-  <meta itemprop="isAccessibleForFree" content="false" />
-  <div
-    itemprop="isPartOf"
-    itemscope
-    itemtype="http://schema.org/CreativeWork http://schema.org/Product"
-  >
-    <meta itemprop="name" content="The Norcal Tribune" />
-    <meta itemprop="productID" content="norcal_tribute.com:basic" />
-  </div>
-</div>
-```
+#### Authorization endpoint
 
-A usable configuration will provide `NewsArticle` typed item with `isAccessibleForFree` property and a subitem of type `Product` that specifies the `productID`.
+Authorization is an endpoint provided by the Publisher and called by the AMP Runtime. It is a credentialed CORS GET endpoint.
 
-In this example:
+**Note:** The Authorization endpoint must implement the security protocol described in the
+[AMP CORS Security Spec](https://amp.dev/documentation/guides-and-tutorials/learn/amp-caches-and-cors/amp-cors-requests#cors-security-in-amp).
 
-1.  The product ID is "norcal_tribune.com:basic" (`"productID": "norcal_tribune.com:basic"`).
-2.  This document is currently locked (`"isAccessibleForFree": false`).
+This endpoint returns the Entitlements object that can be used by the [Attributes][11] to hide or show different parts of the document. Authorization endpoint is specified using the "authorizationUrl" property in the config.
 
-The configuration is resolved as soon as `productID` and `isAccessibleForFree` are found. It is, therefore, advised to place the configuration as high up in the DOM tree as possible.
+The Entitlement response returned by the authorization endpoint must conform to the predefined format:
```

Comment:
What format? Below is a table.

Reply:
Will change "format" to "specification, as defined in the table below"